### ISSUE TITLE
Some questions about Adaptive Attack

### ISSUE THREAD

**Issue by liushuaihe** [zhyblue424/TGA-ZSR#3]
I want to ask whether there has been an update on the experimental design details and code for Adaptive Attack.

**Comment by zhyblue424** [comment]
Hello, thank you for your interest in our work. Detailed descriptions of the adaptive attack design and experimental settings are provided in Appendix B. However, it seems that the open-access version does not include the appendix. We will be uploading the full paper, including the appendix, to arXiv in the near future.

Here, we provide a brief overview: To align with the principle of adaptive attacks, we design attack objectives that are explicitly tailored to the defense strategies of different methods, rather than relying on a single fixed loss. Based on the form of the optimization objective, the adaptive attacks used in our evaluation can be broadly categorized into two types:  Cross-Entropy-based attacks(TeCoA, LAAT) and Log-Sum-Exp-based attacks (FT-Adv., FARE, PMG-AFT, TGA-ZSR, Comp-TGA).

To adhere more closely to the adaptive attack principle, we extend this objective by augmenting the base LSE loss with method-specific terms that directly target the defense mechanisms of each baseline. As part of this initial exploration, the adaptive attacks we initially design for each method may require further refinement. 

Example PyTorch implementation of the Log-Sum-Exp-based loss:

```python
output = (img_embed @ text_embed.t()) * model.logit_scale.exp()
true_logits = output.gather(1, target.view(-1,1)).squeeze(1)
tau = 1.0
masked = output.clone()
masked.scatter_(1, target.view(-1,1), -1e4)
lse_other = torch.logsumexp(masked / tau, dim=1) * tau
loss = (lse_other - true_logits).mean()
```

Finally, the [appendix.pdf](https://github.com/user-attachments/files/25833962/appendix.pdf) is attached for your reference. For any questions, you can reach us at zshy@stud.tjut.edu.cn.

**Comment by liushuaihe** [comment]
Thank you very much for your reply
